### PR TITLE
[Messenger] Activation middleware decorator

### PR DESCRIPTION
**Pull request by ogizanagi**
| Q             | A
| ------------- | ---
| Branch?       | master 
| Bug fix?      | no
| New feature?  | yes 
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | part of #26901   
| License       | MIT
| Doc PR        | TODO: https://github.com/symfony/symfony-docs/issues/10035

A small middleware decorator that can be wired using DI decoration to enable/disable a middleware on an arbitrary condition. This can be used to keep the same middleware stack across env but enable/disable some of them through this.

### CODE REVIEW

**Comment by sroze** [comment]
How would someone use it? Manually, as in registers a service that `decorates` the generated middleware?

**Comment by ogizanagi** [comment]
@sroze : Exactly. It'll just leverage DI capabilities.

File: src/Symfony/Component/Messenger/Middleware/Enhancers/ActivationMiddlewareDecorator.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    public function __construct(MiddlewareInterface $inner, $activated)
+    {
+        $this->inner = $inner;
+        $this->activated = $activated;
+    }
+
+    /**
+     * @param Envelope $message
+     */
+    public function handle($message, callable $next)
```
Comment (sroze) [inline src/Symfony/Component/Messenger/Middleware/Enhancers/ActivationMiddlewareDecorator.php]: PHP allows you to rename `$message` to `$envelope` here, right?

**Review by sroze — APPROVED** [review]
Makes sense to me 👍 (we definitely need a documentation PR for this though :))

File: src/Symfony/Component/Messenger/Middleware/Enhancers/ActivationMiddlewareDecorator.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Middleware\Enhancers;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Messenger/Middleware/Enhancers/ActivationMiddlewareDecorator.php]: Is the `Enhancers` sub-namespace required? Nesting hurts discovery IMHO, better remove it?

**Review by nicolas-grekas — APPROVED** [review]
LGTM, with one naming question.

File: src/Symfony/Component/Messenger/Middleware/Enhancers/ActivationMiddlewareDecorator.php (hunk 2 above)
Comment (ogizanagi) [reply src/Symfony/Component/Messenger/Middleware/Enhancers/ActivationMiddlewareDecorator.php]: This was mainly echoing `Symfony\Component\Messenger\Transport\Enhancers` which are decorators. But of course I'm fine with removing this namespace.

**Comment by kbond** [comment]
Is there a doc PR for this?

**Comment by ogizanagi** [comment]
@kbond : Not yet. Feel free to open one if you want. Otherwise it'll go in my TODO list anyway :)

**Pull request merged**